src/middlewared/middlewared/plugins/gluster_linux/volume.py

```diff
@@ -0,0 +1,542 @@
+
+    @private
+    def list_volumes(self):
+
+        result = self.__volume_wrapper(volume.vollist)
+
+        return result
+
+    @private
+    def optreset_volume(self, data):
+
+        volname = data.pop('volname')
+
+        result = self.__volume_wrapper(volume.optreset, volname, **data)
+
+        return result
+
+    @private
+    def optset_volume(self, data):
+
+        volname = data.pop('volname')
+
+        result = self.__volume_wrapper(volume.optset, volname, **data)
+
+        return result
+
+    @private
+    def addbrick_volume(self, data):
+
+        volname = data.pop('volname')
+        volbricks = data.pop('bricks')
+
+        bricks = []
+        for i in volbricks:
+            peer = i['peer_name']
+            path = i['peer_path']
+            brick = peer + ':' + path
+            bricks.append(brick)
+
+        result = self.__volume_wrapper(
+            volume.bricks.add, volname, bricks, **data)
+
+        return result
+
+    @private
+    def removebrick_volume(self, data):
+
+        volname = data.pop('volname')
+        volbricks = data.pop('bricks')
+        op = data.pop('operation')
+
+        bricks = []
+        for i in volbricks:
+            peer = i['peer_name']
+            path = i['peer_path']
+            brick = peer + ':' + path
+            bricks.append(brick)
+        # TODO
+        # glustercli-python has a bug where if provided the "force"
+        # option, it will concatenate it with the "start" option
+        # This is wrong, you can choose "start" or "force" exclusively
+        # (i.e. gluster volume volname remove-brick peer:path start OR force)
+        if op.lower() == 'start':
+            result = self.__volume_wrapper(
+                volume.bricks.remove_start, volname, bricks, **data)
+
+        if op.lower() == 'stop':
+            result = self.__volume_wrapper(
+                volume.bricks.remove_stop, volname, bricks, **data)
+
+        if op.lower() == 'commit':
+            result = self.__volume_wrapper(
+                volume.bricks.remove_commit, volname, bricks, **data)
+
+        if op.lower() == 'status':
+            result = self.__volume_wrapper(
+                volume.bricks.remove_status, volname, bricks, **data)
+
+        return result
+
+    @private
+    def replacebrick_volume(self, data):
+
+        volname = data.pop('volname')
+        src = data.pop('src_brick')
+        new = data.pop('new_brick')
+
+        src_peer = src['peer_name']
+        src_path = src['peer_path']
+        src_brick = src_peer + ':' + src_path
+
+        new_peer = new['peer_name']
+        new_path = new['peer_path']
+        new_brick = new_peer + ':' + new_path
+
+        result = self.__volume_wrapper(
+            volume.bricks.replace_commit,
+            volname,
+            src_brick,
+            new_brick,
+            **data)
+
+        return result
+
+    @accepts(Dict(
+        'glustervolume_create',
+        Str('volname', required=True),
```

Comment:
Since the namespace already is called gluster.volume, perhaps it could be just "name" and "bricks", removing the suffix? Thoughts?

Reply:
I think I see what you're saying. So instead of `glustervolume_create` change it to `name_create` inside the `@accepts(Dict(`? And then do the same for the brick operations too?

Reply:
No, I was saying `name` instead of `volname`, and `bricks` instead of `volbricks` as attribute/param name

Reply:
Understood, I mostly stuck those names because the called methods use those names. However, most of the time those names didn't matter since the called method uses those parameters as positional args anyways.